### PR TITLE
Fixing FindBugs warnings

### PR DESCRIPTION
**Pull request by chavdar**
Issues fixed:
* gobblin-core/src/main/java/gobblin/source/extractor/utils/Utils.java doing unnecessary boxing/unboxing of long
* gobblin-runtime/src/main/java/gobblin/runtime/Task.java using ++ on a volatile
* gobblin-runtime/src/main/java/gobblin/runtime/local/LocalJobManager.java - negating an integer in compare() (may cause integer overflow)
* gobblin-runtime/src/main/java/gobblin/runtime/local/LocalJobManager.java not closing a stream
* gobblin-utility/src/main/java/gobblin/util/AvroUtils.java - making FB shut up that input can be null

@liyinan926 @ibuenros can you have a look?

### CODE REVIEW

**Comment by sahilTakiar** [comment]
For `AvroUtils` I think you can just tag it with `javax.annotation.Nonnull`. That should make the FindBugs warnings go away.

File: gobblin-core/src/main/java/gobblin/source/extractor/utils/Utils.java [hunk 1]
```diff
@@ -12,6 +12,9 @@
 
 package gobblin.source.extractor.utils;
 
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.source.extractor.watermark.WatermarkType;
```
Comment (liyinan926) [inline gobblin-core/src/main/java/gobblin/source/extractor/utils/Utils.java]: The convention for import orders we use put gobblin imports at the end.

File: gobblin-runtime/src/main/java/gobblin/runtime/Task.java [hunk 2]
```diff
@@ -38,6 +25,20 @@
 import gobblin.qualitychecker.row.RowLevelPolicyChecker;
 import gobblin.runtime.util.RuntimeConstructs;
 
+import java.io.IOException;
```
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/Task.java]: Ditto. We use the ordering throughout the code base.

File: gobblin-runtime/src/main/java/gobblin/runtime/Task.java [hunk 3]
```diff
@@ -82,7 +83,7 @@
   private final List<Optional<Fork>> forks = Lists.newArrayList();
 
   // Number of task retries
-  private volatile int retryCount = 0;
+  private AtomicInteger retryCount = new AtomicInteger();
```
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/Task.java]: This can be final.

Comment (chavdar) [inline gobblin-runtime/src/main/java/gobblin/runtime/Task.java]: Fixed.

**Comment by chavdar** [comment]
@sahilTakiar It is defined as nullable in the Guava interface, this is the contract of the API.

File: gobblin-core/src/main/java/gobblin/source/extractor/utils/Utils.java (hunk 1 above)
Comment (chavdar) [inline gobblin-core/src/main/java/gobblin/source/extractor/utils/Utils.java]: Fixed.

File: gobblin-runtime/src/main/java/gobblin/runtime/Task.java (hunk 2 above)
Comment (chavdar) [inline gobblin-runtime/src/main/java/gobblin/runtime/Task.java]: Fixed.

**Comment by liyinan926** [comment]
LGTM.

**Pull request merged**